### ISSUE TITLE
On lid close, activate the screen saver before suspending

### ISSUE THREAD

**Issue by sirlark** [lxde/lxqt#431]
When the lid is closed on my laptop with an SSD as the OS drive, I can open the lid and come out of suspend (to memory) and I still see my screen as it was before I closed the lid, for about a second before the screen saver/locker kicks in. The screen should lock before the suspend takes effect.

**Comment by surlykke** [comment]
I don't think this is an lxqt-powermanagement issue - or even an lxqt
issue. lxqt-powermanagement calls liblxqt/power to do suspend, and
liblxqt/power, in turn, calls some underlying service. The underlying
service could be upower, systemd, consolekit or something else, depending
on your distro, and init system.

What if you choose suspend from the menu, let your computer suspend, close
the lid and open the lid again - do you see the issue?
What if you suspend from the command line, close lid, open lid?

(on systemd systems I believe you can suspend from the command-line like
'sudo systemctl suspend'. I don't use systemd personally, so I'm not
sure...)

I don't know what activates the screen lock on suspend. I'm pretty sure I
haven't coded such logic in lxqt-powermanagement :-), Maybe it's
xscreensaver that somehow hooks itself into the suspend process...

br. Chr.


2015-01-11 12:47 GMT+01:00 sirlark <notifications@github.com>:

**Comment by sirlark** [comment]
I think the screen lock suspends because after resuming, xscreensaver sees the jump in time. The thing is with and SSD (or any other system that is takes less than a second to to resume), the screen becomes visible before xscreensaver has the chance to realise time has elapsed.

Perhaps it would be useful to have an option to lock the screen on suspend, or is there a hook to do this already?

**Comment by surlykke** [comment]
2015-01-11 13:25 GMT+01:00 sirlark <notifications@github.com>:

> I think the screen lock suspends because after resuming, xscreensaver sees
> the jump in time. The thing is with and SSD (or any other system that is
> takes less than a second to to resume), the screen becomes visible before
> xscreensaver has the chance to realise time has elapsed.
>
> Perhaps it would be useful to have an option to lock the screen on
> suspend, or is there a hook to do this already?
>
> We don't have that. So let's leave this issue open.

br. Chr.

**Comment by ufukkilicaslan** [comment]
Actually, I really don't understand how lxqt locks the screen. There is a lock screen option in power management settings, on idleness watcher, but it does not work for me. I only found a file related to this, which /etc/xdg/autostart/lxqt-screenlocker.desktop I really try to understand how I can lock my screen after idle without using xscreensaver. And if lxqt-screenlocker, or the screen locking mechanism of which lxqt use needs an actual screen locker app, light xscreensaver and light-locker, why there is nothing installed.

If I use xscreensaver, it has its own options regarding the idleness. I am looking for a file or script of which idleness watcher uses as its configuration file. There is power management config in .config/lxqt but there are only choices (like 1, 2, 3, 4, namely for suspend, lock screen, etc). In the /etc/xdg/autostart/lxqt-screenlocker.desktop file, there is a line like Exec=lxqt-screenlocker  But it doesn't point anything.

So, tl;dr: My first objective is to lock screen after some time of idleness, using lxqt power management settings. It does not work for me (using Arch Linux official settings). I am using sddm as login manager, so I don't have light-locker (coming with lightdm). If there is no mechanism or executable which locks the screen in lxqt, how can we use another screensaver? Second objective is, to lock my screen after suspend. In light-locker, there was such option in its settings.

**Comment by ufukkilicaslan** [comment]
Actually, I am wondering how O.P. managed to activate the screensaver (even it kicks in after some time) with using lxqt-powermanagement, with suspend or without suspend. I tried to run xscreensaver daemon, to see if it's related to it, but still lock screen option in power management idleness settings doesn't work. If I set it to suspend, it works (without locking the screen of course).

**Comment by sirlark** [comment]
I have a file xscreensaver in my autostart (~/.config/autostart/razor-xscreensaver-autostart.desktop)

**Comment by polomi** [comment]
Since this is still open, I'll just say that I also think that it would be nice to have an option to call the screensaver when starting suspend.

Also, if Idleness Watcher is enabled in the Power Management Settings while XScreenSaver is not installed, at the time when it tries to call the screensaver it will show a pop-up with an error message saying that it's not installed. Because of this, maybe XScreenSaver should be marked as a dependency of lxqt-powermanagement. It is not even marked as an optional dependency in the Arch repos.

**Comment by sirlark** [comment]
I just upgraded to lxqt-powermanagement-0.10.0 on laptop and now I'm having the same problems as @ufukkilicaslan 

I've resorted to disabling the lid watcher since it no longer seems able to initiate a suspend, and using acpid hooks instead. Idleness watcher can still initiate screen lock though.

**Comment by paulolieuthier** [comment]
@sirlark if you can test the last referenced PR it'd be great.

**Comment by jeremywh7** [comment]
Ever since my last AUR update, I've been getting this upon Suspend:
![1467754467](https://cloud.githubusercontent.com/assets/3794434/16602296/f49f98cc-42e0-11e6-94b5-0d160fe84b45.png)
I don't use a screensaver.  This commit seems the likely culprit...?

**Comment by luis-pereira** [comment]
@jeremywh7 That issue got fixed with https://github.com/lxde/liblxqt/commit/cd10d270b85b6e7cf1382c22bcad5e301a858c97

**Comment by agaida** [comment]
@luis-pereira - sorry, no, it isn't

**Comment by paulolieuthier** [comment]
To me it looks fixed. @jeremywh7, could you please update and test again?

**Comment by luis-pereira** [comment]
@agaida You have to be more specific. with what action ?

**Comment by jeremywh7** [comment]
Does it matter that I did not reboot after the last update?  I exited lxqt,
updated, then relaunched it.  This was several days after the previous
update. I recall only one package plus maybe libfm updated.  Unfortunately
I am out of town on vacation until Sunday, but will check it out when I get
back. Thanks guys!

**Comment by agaida** [comment]
@luis-pereira and @paulolieuthier - you are right, it works as expected. But this one drives me crazy. All i had to do was to delete ~/config/lxqt. And i guess that a few other bugs will be "solved" with deleting user configurations - but thats a wild guess, unfortunately i relly deleted the .config/lxqt - i should have moving it out of the way.

**Comment by tsujan** [comment]
> All i had to do was to delete ~/config/lxqt.

Backward incompatibility to such an extent isn't acceptable, IMHO.

**Comment by agaida** [comment]
sorry - at a first look all seems to be ok - but it isn't. @luis-pereira: first i tested only the lid behaviour - first lid action was ok, second lid action leads to locking screnn after wake up, third lid action brings up the window - but not every time. Once the mentioned window came up any lid action bring this.

After that i tested the lxqt-leave menu, same result. first suspend was ok, the follow up suspends get worse.

**Comment by stefonarch** [comment]
Similar here, I'm using xscreensaver on a Desktop PC as screensaver  but never lock the screen and I find it locked after suspend now, option in screensaver is still unchecked.

**Comment by luis-pereira** [comment]
@tsujan I agree with you, but neither my or @paulolieuthier code touches settings.
@agaida I'm testing it out.

**Comment by agaida** [comment]
@luis-pereira - sorry for being pain in the ass - but what should i do, i would like to track this down myself but unfortunately i have to learn some basics to do so first - but it will be better over time. :)

**Comment by luis-pereira** [comment]
@agaida no pain in the ass.
I tested it. I detected one bug: The Idleness watcher did not always launch the screensaver after the first time. It's fixed by the https://github.com/lxde/lxqt-powermanagement/pull/37 PR. Test it.
Other than that, I did not found any issue. Everything is working as expected. I'm not saying that there aren't issues, all I'm saying is that I can't reproduce them.
I'm running a bare metal Arch. 

@stefonarch With @paulolieuthier code, before an suspend and hibernate we lock the screen. There's no option for that.

**Comment by tsujan** [comment]
> @stefonarch With @paulolieuthier code, before an suspend and hibernate we lock the screen. There's no option for that.

I may be totally lost here but do you mean that screen locking is mandatory now? And what if someone (like me) doesn't use xscreensaver at all?

**Comment by stefonarch** [comment]
> @stefonarch With @paulolieuthier code, before an suspend and hibernate we lock the screen. There's no option for that.

Uhm, actually I would like to decide by myself when locking screen  and when not. 
Is is difficult to add an option "lock screen before suspend/hibernate" in the general settings section in lxqt-config-session ?

**Comment by Vladimir-csp** [comment]
Given that xscreensaver is not the only available locker, custom lock command option would also be a logical addition.

**Comment by tsujan** [comment]
Please don't get me wrong! I agree that screen locking is an important security feature but I can't agree on making it compulsory. Let the user decide whether he/she wants it!

**Comment by luis-pereira** [comment]
@Vladimir-csp I do agree with you. But not before the release.
@stefonarch That's the description for this issue.
@tsujan Maybe adding an Advanced page on the configuration dialog.

**Comment by tsujan** [comment]
> Maybe adding an Advanced page on the configuration dialog. 

IMO, that config would be important -- LXQt has never forced its users into such things, as far as I know.

**Comment by agaida** [comment]
@Vladimir-csp - the implemented locking mechanism depend on xdg-screensaver which is a wrapper around possible screensavers - ok, xscreensaver is the most common and feasable for lxqt i think.
@tsujan - you re right to some extend - if i would use LXQt on a laptop (i do) the only save option before running around is turn the laptop off - one had to do a full boot with login to play it save.

otherwise - we shouldn't hinder users to shoot themself into one or both feets :)

**Comment by tsujan** [comment]
> if i would use LXQt on a laptop (i do) the only save option before running around is turn the laptop off

LOL! That's exactly what I do. So, I've never needed screen locking. And I prefer automatic monitor turn-off rather than screensavers, which I achieve by a simple start-up script in any no-nonsense DE.

> Edit: Second option - do nothing if no screensaver is found - third option: run the screensaver without locking the screen.

I don't know how hard it is to add such options but that's reasonable.

**Comment by agaida** [comment]
third option should be a screensaver thing - also the blank only and turn the off the monitor should be a screensaver setting and should nothing have to do with activation of the xdg-screensaver.

@luis-pereira - i merged your changes, build and install it, unfortunately it changes nothing in the locking behavior - esp. if i use the lxqt-leave dialog.

**Comment by agaida** [comment]
ok - i think i will set up a new installation (debian sid) with the snapshot repos tomorrow. On the desktop suspend and locking works just fine -- ok, i don't use swear words yet, but thats my original razorqt installation from 2013 - and the laptop is a updated 2015 Siduction release. So i guess that some crazy thing is happend - but i don't know which. Really no clue.

**Comment by luis-pereira** [comment]
@agaida Ok. keep us posted.

**Comment by tsujan** [comment]
Today I upgraded LXQt components from the latest git sources under Debian on laptop and encountered this annoying dialog after opening the lid:

![resume](https://cloud.githubusercontent.com/assets/981076/17083264/264f414c-51af-11e6-8f1f-c8caba212b44.png)

Would you please tell me which commit causes it? As I said earlier, I don't use xscreensaver (and don't *want* to use it).

**Comment by paulolieuthier** [comment]
@tsujan: lxde/liblxqt#99 is supposed to have fixed it.

**Comment by tsujan** [comment]
> lxde/liblxqt#99 is supposed to have fixed it.

Apparently It hasn't.

**Comment by tsujan** [comment]
Should a separate report be created for this? Now I see that @jeremywh7 had reported it here too (https://github.com/lxde/lxqt/issues/431#issuecomment-230625354).

**Comment by jeremywh7** [comment]
Speaking of...I did update everything on the 22nd.  It was failing to build the panel at first, which turned out to be due to the Arch AUR git package missing a dependency for liblxqt (best I could tell at least, I posted about it [here](https://aur.archlinux.org/packages/lxqt-panel-git/)), which was rather old.  However, everything is up to date now, and still have this issue.

**Comment by paulolieuthier** [comment]
@tsujan: do you have xscreensaver installed and running?

**Comment by luis-pereira** [comment]
@paulolieuthier Do we account for the possibility of xscreensaver isn't installed ?

p.s. Right now I can't do any real work. So I'm gathering info for later on.

**Comment by paulolieuthier** [comment]
We simply [print the message](https://github.com/lxde/liblxqt/blob/master/lxqtscreensaver.cpp#L140) for now.

**Comment by luis-pereira** [comment]
@paulolieuthier That feature is not being considered to the next release. We already have enough on our  plate.

**Comment by paulolieuthier** [comment]
@luis-pereira I don't follow you. What feature?

**Comment by luis-pereira** [comment]
Not allowing the user to select Lock if no screensaver is installed.

**Comment by paulolieuthier** [comment]
You're right.

**Comment by tsujan** [comment]
> @tsujan: do you have xscreensaver installed and running?

No, as I said above. And that's exactly the problem: https://github.com/lxde/lxqt-session/pull/70 somehow presupposes the existence of xscreensaver, while the user has right not to want things like screensaver or screen locking -- the DE shouldn't enforce them on him.

**Comment by paulolieuthier** [comment]
Yes, you're right. Just to make it clear: lxde/liblxqt#99 is not supposed to fix this, I didn't understand your issue at first. I'm working to fix it now.

**Comment by tsujan** [comment]
@paulolieuthier Thanks!

**Comment by paulolieuthier** [comment]
Everything related to screen locking on suspeding and on closing the lid is working for me now. Can we close this?

**Comment by tsujan** [comment]
> Can we close this?

Yes, IMO. Here, everything is OK too.

**Comment by All3xJ** [comment]
> All i had to do was to delete ~/.config/lxqt. 

this fixed for me. I had the bug that if I manually click on suspend in the start menu it woks, but if I close the lid of my laptop it doesn't  suspend on Lubuntu 18.04 (with lxqt installed instead of lxde), even though it was set under xfce settings manager.

Anyways deleting ~/.config/lxqt fixed it, even though I lose all the settings and configurations
